Lib/email/policy.py

```diff
@@ -140,11 +140,11 @@ def header_store_parse(self, name, value):
         """
         if hasattr(value, 'name') and value.name.lower() == name.lower():
             return (name, value)
-        if isinstance(value, str) and len(value.splitlines())>1:
-            # XXX this error message isn't quite right when we use splitlines
-            # (see issue 22233), but I'm not sure what should happen here.
+        if (isinstance(value, str) and
+            linesep_splitter.search(self.header_factory(name, value))
+        ):
             raise ValueError("Header values may not contain linefeed "
-                             "or carriage return characters")
+                         "or carriage return characters")
```

Comment:
I would keep the formatting consistent:
```suggestion
                             "or carriage return characters")
```